src/main/java/com/unstoppabledomains/resolution/ResolutionBuilder.java

```diff
@@ -0,0 +1,242 @@
+import com.unstoppabledomains.resolution.naming.service.ZNS;
+import com.unstoppabledomains.resolution.naming.service.uns.UNS;
+import com.unstoppabledomains.resolution.naming.service.uns.UNSConfig;
+import com.unstoppabledomains.resolution.naming.service.uns.UNSLocation;
+
+public class ResolutionBuilder {
+    private static final String INFURA_URL = "https://%s.infura.io/v3/%s";
+    static final String ENS_DEFAULT_URL = "https://mainnet.infura.io/v3/d423cf2499584d7fbe171e33b42cfbee";
+    static final String UNS_DEFAULT_URL = "https://mainnet.infura.io/v3/e0c0cb9d12c440a29379df066de587e6";
+    static final String UNS_L2_DEFAULT_URL = "https://polygon-mumbai.infura.io/v3/e0c0cb9d12c440a29379df066de587e6";
+    static final String ZILLIQA_DEFAULT_URL = "https://api.zilliqa.com";
+
+    static final String ENS_DEFAULT_REGISTRY_ADDRESS = "0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e";
+    static final String ZNS_DEFAULT_REGISTRY_ADDRESS = "0x9611c53BE6d1b32058b2747bdeCECed7e1216793";
+
+    private final Map<NamingServiceType, NSConfig> serviceConfigs;
+    private final Map<UNSLocation, NSConfig> unsConfigs;
+    private IProvider provider;
+    private Resolution.ResolutionBuilderConnector connector;
+
+    public ResolutionBuilder(Resolution.ResolutionBuilderConnector connector) {
+        this.connector = connector;
+        serviceConfigs = new HashMap<>();
+        serviceConfigs.put(NamingServiceType.ZNS, new NSConfig(Network.MAINNET, ZILLIQA_DEFAULT_URL, ZNS_DEFAULT_REGISTRY_ADDRESS));
+        serviceConfigs.put(NamingServiceType.ENS, new NSConfig(Network.MAINNET, ENS_DEFAULT_URL, ENS_DEFAULT_REGISTRY_ADDRESS));
+        
+
+        String unsProxyAddress = NetworkConfigLoader.getContractAddress(Network.MAINNET, "ProxyReader");
+        String unsl2ProxyAddress = NetworkConfigLoader.getContractAddress(Network.MUMBAI_TESTNET, "ProxyReader");
+        unsConfigs = new HashMap<>();
+        unsConfigs.put(UNSLocation.Layer1, new NSConfig(Network.MAINNET, UNS_DEFAULT_URL, unsProxyAddress));
+        unsConfigs.put(UNSLocation.Layer2, new NSConfig(Network.MUMBAI_TESTNET, UNS_L2_DEFAULT_URL, unsl2ProxyAddress));
+        
+        provider = new DefaultProvider();
+    }
+
+    /**
+     * @param nsType  the naming service for which config is applied
+     * @param chainId blockchain network ID
+     * @return builder object to allow chaining
+     */
+    public ResolutionBuilder chainId(NamingServiceType nsType, Network chainId) {
+        NSConfig nsConfig = serviceConfigs.get(nsType);
+        nsConfig.setChainId(chainId);
+        return this;
+    }
+
+    /**
+     * Applies blockchain network ID for UNS naming service 
+     * @param location the location of the UNS service (layer 1 or layer 2)
+     * @param chainId blockchain network ID
+     * @return builder object to allow chaining
+     */
+    public ResolutionBuilder chainId(UNSLocation location, Network chainId) {
```

Comment:
Being a switcher from the languages where such kind of overloading was intentionally made impossible, I'm biased towards the `unsChainId` variant. Though, since I'm also unfamiliar with our Java library, I'm having trouble understanding the difference between these two methods. Are they configuring different services (i.e. `NamingServiceType nsType` variant is for CNS and `UNSLocation location` - for UNS)? Or do they do practically the same, but just using different types of arguments to identify which service to configure?

Depending on which case it is exactly, I'd probably consider other naming and improving documentation.

Reply:
Yes they are configuring different services. The `NamingServiceType nsType` variant is for ZNS and ENS. And `UNSLocation location` is for UNS layer 1 and 2.

My reasoning was that since these methods do essentially the same thing (and in some cases have the same implementation), they can be overloaded. 

As an alternative, we can add a separate enum, just for the builder. So instead of `NamingServiceType` we can have `NamingServiceConfigType` which would have values `ENS, ZNS, UNSL1, UNSL2`. This will allow us to keep a single mapping for configs and simplify builder API and logic while also being clear for users. The only downside that I see is that the builder API will change slightly for new enums.

Reply:
@DancingAxolotl 

Personally I would prefer adding getter VS adding additional enums. 

@sudoryan @JohnnyJumper @hivtushok I would love to see yours opinion

Reply:
I like overloading as well and prefer it over unsMethodName()

Reply:
I've renamed the methods. After thinking about it and writing some tests for the builder, I agree that it's clearer this way.
For example, if the user accidentally writes:
```
 builder.chainId(NamingServiceType.UNS, Network.RINKEBY)
```
It will throw a null pointer exception since the map doesn't have an entry for UNS. And it would be annoying to debug. With separate methods for UNS it should be easier to avoid this error. 

Thanks for everyone's input!